hoomd/md/ActiveForceConstraintComputeGPU.h

```diff
@@ -0,0 +1,245 @@
+// Copyright (c) 2009-2021 The Regents of the University of Michigan
+// This file is part of the HOOMD-blue project, released under the BSD 3-Clause License.
+
+// Maintainer: joaander
+
+#include "ActiveForceComputeGPU.cuh"
+#include "ActiveForceConstraintCompute.h"
+#include "ActiveForceConstraintComputeGPU.cuh"
+
+/*! \file ActiveForceConstraintComputeGPU.h
+    \brief Declares a class for computing active forces on the GPU
+*/
+
+#ifdef __HIPCC__
+#error This header cannot be compiled by nvcc
+#endif
+
+#include <pybind11/pybind11.h>
+
+#ifndef __ACTIVEFORCECONSTRAINTCOMPUTE_GPU_H__
+#define __ACTIVEFORCECONSTRAINTCOMPUTE_GPU_H__
+
+#include <vector>
+namespace py = pybind11;
+using namespace std;
+
+//! Adds an active force to a number of particles with confinement on the GPU
+/*! \ingroup computes
+ */
+template<class Manifold>
+class PYBIND11_EXPORT ActiveForceConstraintComputeGPU
+    : public ActiveForceConstraintCompute<Manifold>
+    {
+    public:
+    //! Constructs the compute
+    ActiveForceConstraintComputeGPU(std::shared_ptr<SystemDefinition> sysdef,
+                                    std::shared_ptr<ParticleGroup> group,
+                                    Scalar rotation_diff,
+                                    Manifold manifold,
+                                    Scalar deltaT);
+
+    protected:
+    unsigned int m_block_size; //!< block size to execute on the GPU
```

Comment:
Use an Autotuner in each of `ActiveForceComputeGPU` and `ActiveForceConstraintComputeGPU<Manifold>`. Different GPUs, system sizes, etc... lead to different optimal block sizes. `Autotuner` finds the fastest block size.